pwiz_tools/Skyline/SettingsUI/Irt/CalibrateIrtDlg.cs

```diff
@@ -692,46 +693,67 @@ private static List<MeasuredPeptide> FindBestPeptides(SrmDocument doc, int pepti
-                 * the number of peptides asked for.
-                 * 
-                 * It does this by considering peptides 3 at a time: (prev, current, next) triplets.
-                 * When the pointer has shifted so that the "current" peptide is closer than
-                 * either of its neighbors, then that peptide is added to the standard and removed
-                 * from the search list.
-                 */
-                List<MeasuredPeptide> standardPeptides = new List<MeasuredPeptide>();
-                double gradientLength = maxRT - minRT;
-                for (int i = 0; i < peptideCount; i++)
-                {
-                    double targetRT = minRT + i * (gradientLength / (peptideCount - 1));
-                    for (int j = 0; j < docPeptides.Count; j++)
+                var resultsHandler = new MProphetResultsHandler(doc, model);
+                resultsHandler.ScoreFeatures(null, true);
+
+                var docTargets = docPeptides.ToDictionary(pep => pep.Target, pep => pep);
+                var docNodePeps = doc.Molecules.Where(nodePep => docTargets.ContainsKey(nodePep.ModifiedTarget)).ToArray();
+                var scoredPeptides = new List<ScoredPeptide>(docNodePeps.Length);
+                foreach (var nodePep in docNodePeps)
+                {
+                    var allStats = doc.MeasuredResults.MSDataFileInfos
+                        .Select(info => resultsHandler.GetPeakFeatureStatistics(nodePep.Id.GlobalIndex, info.FileId.GlobalIndex))
+                        .Where(stats => stats != null).ToArray();
+                    var value = float.MaxValue;
+                    if (allStats.Length > 0)
                     {
-                        if (j + 1 > docPeptides.Count - 1 ||
-                                Math.Abs(docPeptides[j].RetentionTime - targetRT) <
-                                Math.Abs(docPeptides[j + 1].RetentionTime - targetRT))
-                        {
-                            standardPeptides.Add(docPeptides[j]);
-                            docPeptides.RemoveAt(j);
-                            break;
-                        }
+                        value = model is MProphetPeakScoringModel
+                            ? allStats.Select(stats => stats.QValue.Value).Max()
+                            : -allStats.Select(stats => stats.BestScore).Min();
                     }
+                    scoredPeptides.Add(new ScoredPeptide(docTargets[nodePep.ModifiedTarget], value));
                 }
 
-                return standardPeptides;
+                var retentionTimes = scoredPeptides.Select(pep => pep.Peptide.RetentionTime).OrderBy(rt => rt).ToArray();
+                var minRt = retentionTimes.First();
+                var maxRt = retentionTimes.Last();
+                var rtRange = maxRt - minRt;
+
+                var buckets = PeptideBucket.BucketPeptides(scoredPeptides, new[]

```

Comment:
I would have probably done something with "IEnumerable.ToLookup" instead of making a separate PeptideBucket class, but this works too.